### PR TITLE
Fix NSIS running twice

### LINKED ISSUE #2110: Installer runs twice
While running the new builder installer I noticed that Tribler (and possibly also the old installer) runs the installation and uninstallation section twice. This is confusing and has to be investigated :)

### PR DESCRIPTION
**Pull request by lfdversluis**
So after quite a while I managed to get rid of the double running problem. The problem was in the UAC plugin and after having a nice chat with the author Anders (nice guy by the way :D) he told me to sacrifice the check box at the last page and get rid of the UAC plugin.

So I got rid of the UAC and kept the check box anyway that now calls the desktop shortcut that is created during installation and works perfect. Tested on proxmox 144 and works splendid.
While I was at it, I formatted the file because it was 2, 3 and 4 space indented mixed with tabs.
Also added a bit of comments to let future generations know why decisions were made and what it does.

Oh... ignore most fixup commits as they feature tries by me to either identify the root cause of this bug, get to know how stuff works, debug items or simply to test things.

Fixes **1** blocker bug
Comments and formatting improved (enhancement)
Got rid of UAC plugin (enhancement)

Fixes #2110

### CODE REVIEW

**Comment by lfdversluis** [comment]
@devos50 can you review?

File: Tribler/Main/Build/Win/tribler.nsi [hunk 1]
```diff
@@ -110,179 +109,178 @@ LangString DESC_SecDefaultMagnet ${LANG_ENGLISH} "Associate magnet links with ${
...
+    ; Now writing to KHEY_LOCAL_MACHINE only -- remove references to uninstall from current user
+    DeleteRegKey HKEY_CURRENT_USER "SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\${PRODUCT}"
+    ; Remove old error log if present
+    Delete "$INSTDIR\tribler.exe.log"
+
+    WriteUninstaller "$INSTDIR\Uninstall.exe"
+
+    ; Add an application to the firewall exception list - All Networks - All IP Version - Enabled
+    SimpleFC::AddApplication "Tribler" "$INSTDIR\${PRODUCT}.exe" 0 2 "" 1
+    ; Pop $0 ; return error(1)/success(0)
```
Comment (devos50) [inline Tribler/Main/Build/Win/tribler.nsi]: If we are not using this code, maybe we should just remove it? I saw some more occurrences of commented-out code.

File: Tribler/Main/Build/Win/tribler.nsi [hunk 2]
```diff
@@ -325,72 +322,39 @@ SectionEnd
...
-Function un.onInit
-	SetShellVarContext all
-    	!insertmacro Init "uninstaller"
 FunctionEnd
 
 Function PageFinishRun
-	!insertmacro UAC_AsUser_ExecShell "" "$INSTDIR\tribler.exe" "" "" ""
+    ; This function is invoked if the user leaves the checkbox checked at the last page (do you wish to run...)
+    ; It calls the freshly created desktop shortcut
+    ExecShell "" "$DESKTOP\${PRODUCT}.lnk"
```
Comment (devos50) [inline Tribler/Main/Build/Win/tribler.nsi]: Is a desktop shortcut always created? Is it possible to run `tribler.exe` immediately from the installed directory?

**Comment by devos50** [comment]
Nice to see this one solved! Really annoying to have a bug in a plugin that we are using...

In the future, can you squash the commits first before I review? That makes it a little easier for me to review by commit (so I can distinguish better between bug fixes and formatting fixes).

File: Tribler/Main/Build/Win/tribler.nsi (hunk 2 above)
Comment (lfdversluis) [inline Tribler/Main/Build/Win/tribler.nsi]: yes the shortcut is always created. As far as I am aware not, as it was ran with admin permission previously.

**Comment by lfdversluis** [comment]
@whirm prefers to leave the fixups, at least when a review has happened. Before it may be ok to squash

File: Tribler/Main/Build/Win/tribler.nsi (hunk 1 above)
Comment (lfdversluis) [inline Tribler/Main/Build/Win/tribler.nsi]: Legacy stuff. I guess it can go now, yes.

**Comment by synctext** [comment]
admin level access level... Does that alter the Windows install experience? 

noce fix!

**Comment by lfdversluis** [comment]
@synctext No, currently users will also see the "require access" screen because we are touching the restricted C:/ folders. The situation will only improve now as previously two processes ran (which you also saw in the taskmanager), now it's down to one :)

**Comment by whirm** [comment]
> @whirm prefers to leave the fixups, at least when a review has happened. Before it may be ok to squash

I prefer to have fixup commits while I'm reviewing. If you want to be extra nice to each other, you guys can do a squash before @devos50 / @lfdversluis reviews and another before I review.

**Comment by lfdversluis** [comment]
@whirm squashed, ready for your review.

File: Tribler/Main/Build/Win/tribler.nsi [hunk 3]
```diff
@@ -1,16 +1,15 @@
 !define PRODUCT "Tribler"
-; Laurens, 2016-03-14: The __GIT__ string will be replaced by update_version_from_git.py
+; Laurens, 2016-03-14: The __GIT__ string will be replaced by update_version_from_git.py with the current version of the build.
```
Comment (whirm) [inline Tribler/Main/Build/Win/tribler.nsi]: Please, 120 cols if possible.

File: Tribler/Main/Build/Win/tribler.nsi [hunk 4]
```diff
@@ -110,179 +109,175 @@ LangString DESC_SecDefaultMagnet ${LANG_ENGLISH} "Associate magnet links with ${
 ;Installer Sections
 
 Section "!Main EXE" SecMain
- SectionIn RO
```
Comment (whirm) [inline Tribler/Main/Build/Win/tribler.nsi]: Please, no reformatting for untouched code

**Comment by whirm** [comment]
Have you tested it on win 7, 8, 10?

**Comment by lfdversluis** [comment]
Only on win10, do we don't have proxmox machines running 7 and 8. I can test it on my win 7, but then 8 is still untested. So either I need to install a VM or a proxmox with 8 :D

File: Tribler/Main/Build/Win/tribler.nsi (hunk 4 above)
Comment (lfdversluis) [inline Tribler/Main/Build/Win/tribler.nsi]: So I need to undo all the identation mess fixes for lines I didn't modify? :/

**Comment by whirm** [comment]
> So I got rid of the UAC and kept the check box anyway that now calls the desktop shortcut that is created during installation and works perfect.

Does it run Tribler as Admin?

**Comment by lfdversluis** [comment]
Good question, have to investigate that one. I will do this in the evening today.

**Comment by lfdversluis** [comment]
It ran under my user, so I assume user rights, couldn't see a setting admin vs user permission though. Tested on win 7 so only win 8 to verify.

**Comment by lfdversluis** [comment]
@whirm I removed the long comment lines > 120 chars. The refactoring is in a separate commit as discussed yesterday. Do you have a win-8 that can test the last build of May 6? You should see some alertboxes if you install, those are gone now.

**Comment by whirm** [comment]
> It ran under my user, so I assume user rights, couldn't see a setting admin vs user permission though. Tested on win 7 so only win 8 to verify.

Look at the process manager, if Tribler gets executed as admin when the install finishes, that would be really bad, and that's probably what's happening if the author told you about losing the checkbox at the end.

**Comment by lfdversluis** [comment]
Found the option to enable UAC in the manager. It is indeed running as admin. I propose to let the checkbox go then as Anders argued.

**Comment by whirm** [comment]
Ok

**Comment by lfdversluis** [comment]
Removed the run page, tested it on win 7 and 10

**Comment by lfdversluis** [comment]
@whirm Could you test the installer on a win 8 machine? You can grab it here: https://jenkins.tribler.org/job/Builder_64_laurens/

**Comment by whirm** [comment]
I don't have a windows 8. only vista and 10.

**Comment by whirm** [comment]
I don't have a windows 8. only vista and 10.

**Comment by lfdversluis** [comment]
https://jenkins.tribler.org/job/pers/job/Builder_64_laurens/ now

**Comment by lfdversluis** [comment]
Works on win 8

**Comment by whirm** [comment]
Great!, can you squash it?

**Pull request merged**